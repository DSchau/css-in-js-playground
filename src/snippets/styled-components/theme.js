export default {
  base: '#f6f9fc',
  header: '#ddecf7'
}
