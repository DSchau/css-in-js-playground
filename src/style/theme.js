// @flow
type StyleBase = {
  base: string,
  text: string
};

export type Theme = {
  dark: StyleBase,
  light: StyleBase,
  primary: 'dark' | 'light'
};

export type ThemeProps = {
  theme?: Theme
};

export const THEME = {
  dark: {
    base: '#1E1E20',
    text: '#ddd'
  },
  light: {
    base: '#ECF0F1',
    text: '#1E1E20'
  },
  primary: 'dark'
};
