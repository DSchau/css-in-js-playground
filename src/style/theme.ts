interface StyleBase {
  base: string;
  text: string;
}

export interface Theme {
  dark: StyleBase;
  light: StyleBase;
  primary: 'dark' | 'light';
}

export const THEME = {
  dark: {
    base: '#1E1E20',
    text: '#ddd'
  },
  light: {
    base: '#ECF0F1',
    text: '#1E1E20'
  },
  primary: 'dark'
};
