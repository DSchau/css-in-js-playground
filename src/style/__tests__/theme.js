import { THEME } from '../theme';

test('it contains a THEME object', () => {
  expect(THEME).toMatchSnapshot();
});
