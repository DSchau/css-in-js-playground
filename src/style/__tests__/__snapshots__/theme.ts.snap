// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it contains a THEME object 1`] = `
Object {
  "dark": Object {
    "base": "#1E1E20",
    "text": "#ddd",
  },
  "light": Object {
    "base": "#ECF0F1",
    "text": "#1E1E20",
  },
  "primary": "dark",
}
`;
